File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.py
class Solution:
    def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
        unique_x = set()
        for point in points:
            unique_x.add(point[0])

        sorted_x = sorted(unique_x)

        max_width = 0

        for i in range(1, len(sorted_x)):
            current_gap = sorted_x[i] - sorted_x[i - 1]
            if current_gap > max_width:
                max_width = current_gap
        
        return max_width

File: 1742-widest-vertical-area-between-two-points-containing-no-points/widest-vertical-area-between-two-points-containing-no-points.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var maxWidthOfVerticalArea = function(points) {
    points.sort((a, b) => a[0] - b[0]);
    let maxWidth = 0;
    for (let i = 1; i < points.length; i++){
        let width = points[i][0] - points[i - 1][0];
        if (width > maxWidth){
            maxWidth = width;
        }
    }
    return maxWidth;
};